### PR TITLE
Remove loader-specific args from train.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/mx/model/estimator.py [hunk 1]
```diff
@@ -151,11 +156,6 @@ def train_model(
 
         training_data_loader = self.create_training_data_loader(
             transformed_training_data
-            if not cache_data
-            else Cached(transformed_training_data),
-            num_workers=num_workers,
-            num_prefetch=num_prefetch,
-            shuffle_buffer_length=shuffle_buffer_length,
```
Comment (lostella) [inline src/gluonts/mx/model/estimator.py:158]: These can actually be updated to use `loader_args` right?

Comment (jaheba) [reply src/gluonts/mx/model/estimator.py:158]: Just need to find where they are used.

File: src/gluonts/mx/model/estimator.py [hunk 2]
```diff
@@ -35,6 +35,39 @@
...
+        return TrainDataLoader(
+            dataset=dataset,
+            transform=transform,
+            batch_size=batch_size,
+            stack_fn=stack_fn,
+            decode_fn=decode_fn,
+            num_workers=self.num_workers,
+            num_prefetch=self.num_prefetch,
+            shuffle_buffer_length=self.shuffle_buffer_length,
+        )
```
Comment (lostella) [inline src/gluonts/mx/model/estimator.py:57]: Would it make sense to directly do here what happens in `TrainDataLoader`?

Also, this overloads the "data loader" terminology a bit, would it make sense to rename this class to something else, like `DataLoaderConfig` or something similar?

Comment (lostella) [reply src/gluonts/mx/model/estimator.py:57]: For the first point: that would mean getting rid of these functions https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/dataset/loader.py#L226-L377 but we can also keep them around
